certbot-apache/certbot_apache/configurator.py

```diff
@@ -871,6 +893,52 @@ def _create_vhost(self, path):
         self._add_servernames(vhost)
         return vhost
 
+    def _create_vhost_v2(self, node):
+        """Used by get_virtual_hosts to create vhost objects using ParserNode
+        interfaces.
+
+        :param interfaces.BlockNode node: The BlockNode object of VirtualHost block
+
+        :returns: newly created vhost
+        :rtype: :class:`~certbot_apache.obj.VirtualHost`
+        """
+        addrs = set()
+        for param in node.parameters:
+            addrs.add(obj.Addr.fromstring(param))
+
+        is_ssl = False
+        sslengine = node.find_directives("SSLEngine")
+        if sslengine:
+            for directive in sslengine:
+                # TODO: apache-parser-v2
+                # This search should be made wiser. (using other identificators)
+                if directive.has_parameter("on", 0):
+                    is_ssl = True
+
+        # "SSLEngine on" might be set outside of <VirtualHost>
+        # Treat vhosts with port 443 as ssl vhosts
+        for addr in addrs:
+            if addr.get_port() == "443":
+                is_ssl = True
+
+        filename = node.get_filename()
+
+        if filename is None:
+            return None
+
+        macro = False
+        if node.find_directives("Macro"):
+            macro = True
+
+        vhost_enabled = self.parser.parsed_in_original(filename)
+
+        vhost = obj.VirtualHost(filename, node.get_metadata("augeas_path"),
```

Comment:
How would this work if `node` doesn't use Augeas?

Reply:
The augeas path in the VirtualHost object will end up being `None`. This is not used internally by VirtualHost object (for anything else than printing out its `str` representation basically), but is instead used by many helper functions down the line in the `ApacheConfigurator`. 

We'll need to refactor these helper functions regardless of the approach we take here, and this feels like the natural way to handle the issue while in the process of migration.

Reply:
I think I agree. I was trying to avoid the ugly private accesses all around, but I think this makes sense. I'll adjust this and the dependency PR accordingly.